src/algebra/order.lean

```diff
@@ -212,6 +212,21 @@ theorem compares.inj [preorder α] {o₁} :
 | eq a b h₁ h₂ := h₁.eq_eq.2 h₂
 | gt a b h₁ h₂ := h₁.eq_gt.2 h₂
 
+theorem compares_iff_of_compares_impl {β : Type*} [linear_order α] [preorder β] {a b : α}
+  {a' b' : β} (h : ∀ {o}, compares o a b → compares o a' b') (o) :
+  compares o a b ↔ compares o a' b' :=
+begin
+  refine ⟨h, λ ho, _⟩,
+  cases lt_trichotomy a b with hab hab,
```

Comment:
What about
```
rcases lt_trichotomy a b with hab|hab|hab,
```

Reply:
AFAIR, I used `cases` because `rcases` was not available. Should I `import` it?

Reply:
I would leave that for a future reorganization PR. This is a pretty early file.

Reply:
I had missed it wasn't already there.